--- a/src/core/psxinterpreter.cc
+++ b/src/core/psxinterpreter.cc
@@ -758,7 +758,7 @@
     m_regs.pc -= 4;
     exception(Exception::Break, m_inDelaySlot);
     if (m_inDelaySlot) {
-        auto &delayedLoad = m_delayedLoadInfo[m_currentDelayedLoad];
+        auto &delayedLoad = m_delayedLoadInfo[m_currentDelayedLoad ^ 1];
         if (!delayedLoad.pcActive) abort();
         delayedLoad.pcActive = false;
     }
@@ -768,7 +768,7 @@
     m_regs.pc -= 4;
     exception(Exception::Syscall, m_inDelaySlot);
     if (m_inDelaySlot) {
-        auto &delayedLoad = m_delayedLoadInfo[m_currentDelayedLoad];
+        auto &delayedLoad = m_delayedLoadInfo[m_currentDelayedLoad ^ 1];
         if (!delayedLoad.pcActive) abort();
         delayedLoad.pcActive = false;
     }
